### PR TITLE
feat: cycle_check_undirected_graph.cpp

### PR DESCRIPTION
**Pull request by bashishjha**
#### Checklist

- [x] Added description of change
- [x] Added file name matches [File name guidelines](https://github.com/TheAlgorithms/C-Plus-Plus/blob/master/CONTRIBUTING.md#New-File-Name-guidelines)
- [x] Added tests and example, test must pass
- [x] Added documentation so that the program is self-explanatory and educational - [Doxygen guidelines](https://www.doxygen.nl/manual/docblocks.html)
- [x] Relevant documentation/comments is changed or added
- [x] PR title follows semantic [commit guidelines](https://github.com/TheAlgorithms/C-Plus-Plus/blob/master/CONTRIBUTING.md#Commit-Guidelines)
- [x] Search previous suggestions before making a new one, as yours may be a duplicate.
- [x] I acknowledge that all my contributions will be made under the project's license.

Notes: Program to  find cycle in undirected graph using Union find algorithm . Return true if there is cycle in the graph else return false.

<a href="https://gitpod.io/#https://github.com/TheAlgorithms/C-Plus-Plus/pull/1498"></a>

### CODE REVIEW

**Review by Panquesito7 — CHANGES_REQUESTED** [review]
Code is not up to the [repository standards](https://github.com/TheAlgorithms/C-Plus-Plus/blob/master/CONTRIBUTING.md).
Please read them **carefully** and **follow them**.

**Review by Panquesito7 — CHANGES_REQUESTED** [review]
Please fix [`clang-tidy`](https://github.com/TheAlgorithms/C-Plus-Plus/pull/1498/checks?check_run_id=2703409737) warnings.

File: graph/cycle_check_undirected_graph.cpp [hunk 1]
```diff
@@ -0,0 +1,132 @@
...
+             * hence call the union1 method and return false.
+             */
+            union1(u, v);
+            return false;
+      }
+};
+
+/**
+ * Main function
+ */
```
Comment (Panquesito7) [inline graph/cycle_check_undirected_graph.cpp:100]:
```suggestion
/**
 * @brief Main function
 * @returns 0 on exit
 */
```

File: graph/cycle_check_undirected_graph.cpp [hunk 2]
```diff
@@ -0,0 +1,132 @@
...
+ * @brief  Disjoint Set (Or Union-Find) algorithm to check for cycle in a undirected graph.
+ *
+ * @author [Ashish Jha](mailto:b.ashishjha13@gmail.com)
+ *
+ */
+
+#include <iostream> // for std::cout
+#include <utility>  // for std::pair
+#include <vector>   // for std::vector
+
```
Comment (Panquesito7) [inline graph/cycle_check_undirected_graph.cpp:13]:
```suggestion
#include <iostream> /// for IO operations
#include <utility>  /// for std::pair
#include <vector>   /// for std::vector
```

File: graph/cycle_check_undirected_graph.cpp [hunk 3]
```diff
@@ -0,0 +1,132 @@
+/**
+ * @file cycle_check_undirected graph.cpp
+ *
+ * @brief  Disjoint Set (Or Union-Find) algorithm to check for cycle in a undirected graph.
```
Comment (Panquesito7) [inline graph/cycle_check_undirected_graph.cpp:4]: Provide a Wikipedia link in Markdown format (if available/possible) and proper documentation.
If there's no Wikipedia link, add another web source for algorithm explanation. 🙂

Comment (bashishjha) [reply graph/cycle_check_undirected_graph.cpp]: I have added the Wikipedia link .

File: graph/cycle_check_undirected_graph.cpp [hunk 4]
```diff
@@ -0,0 +1,135 @@
+/**
+ * @file cycle_check_undirected graph.cpp
```
Comment (Panquesito7) [inline graph/cycle_check_undirected_graph.cpp:2]:
```suggestion
 * @file
```

File: graph/cycle_check_undirected_graph.cpp [hunk 5]
```diff
@@ -0,0 +1,135 @@
...
+ *
+ * @brief  [Disjoint Set (Or Union-Find) algorithm](https://en.wikipedia.org/wiki/Disjoint-set_data_structure) check for cycle in a undirected graph.
+ *
+ * @author [Ashish Jha](mailto:b.ashishjha13@gmail.com)
```
Comment (Panquesito7) [inline graph/cycle_check_undirected_graph.cpp:6]:
```suggestion
 * @author [Ashish Jha](https://github.com/bashishjha) 
```

File: graph/cycle_check_undirected_graph.cpp [hunk 6]
```diff
@@ -0,0 +1,135 @@
...
+ * @file cycle_check_undirected graph.cpp
+ *
+ * @brief  [Disjoint Set (Or Union-Find) algorithm](https://en.wikipedia.org/wiki/Disjoint-set_data_structure) check for cycle in a undirected graph.
+ *
+ * @author [Ashish Jha](mailto:b.ashishjha13@gmail.com) 
+ * 
+ *
+ */
+
+#include <iostream> // for std::IO operations
```
Comment (Panquesito7) [inline graph/cycle_check_undirected_graph.cpp:11]:
```suggestion
#include <iostream> // for IO operations
```

File: graph/cycle_check_undirected_graph.cpp [hunk 7]
```diff
@@ -0,0 +1,135 @@
+/**
+ * @file cycle_check_undirected graph.cpp
+ *
+ * @brief  [Disjoint Set (Or Union-Find) algorithm](https://en.wikipedia.org/wiki/Disjoint-set_data_structure) check for cycle in a undirected graph.
```
Comment (Panquesito7) [inline graph/cycle_check_undirected_graph.cpp:4]: Please add a detailed description and a better brief description (see, for example, the [`n_queens`](https://github.com/TheAlgorithms/C-Plus-Plus/blob/master/backtracking/n_queens.cpp) algorithm).

**Review by Panquesito7 — CHANGES_REQUESTED** [review]

**Pull request closed without merging**